src/overmapbuffer.cpp

```diff
@@ -715,6 +715,86 @@ tripoint overmapbuffer::find_closest( const tripoint &origin, const std::string
     return overmap::invalid_tripoint;
 }
 
+cata::optional<tripoint> overmapbuffer::find_closest( const tripoint &origin,
+        const std::string &type,
+        int const radius, bool must_be_seen, bool allow_subtype_matches,
+        const std::unordered_set<tripoint> &visited_point )
+{
+    // Check the origin before searching adjacent tiles!
+    if( allow_subtype_matches
+        ? check_ot_subtype( type, origin.x, origin.y, origin.z )
+        : check_ot_type( type, origin.x, origin.y, origin.z ) ) {
+        if( visited_point.find( origin ) == visited_point.end() ) {
+            return origin;
+        }
+    }
+
+    // Function overloading for NPC pathfinding
+    // visited_point is added to check points NPC has been, if any of them matches function ignores the location
+
+    int max = ( radius == 0 ? OMAPX * 5 : radius );
```

Comment:
Seems like are re-implementing the behavior of `closest_tripoints_first`. Consider using that function instead.